Build fails on Debian Bullseye

## Environment
* OS Version: Debian Bullseye
* Source or binary build?
Source build, version 11

## Description
* Expected behavior: The build should pass
* Actual behavior: The build fails

This seems to be similar to https://github.com/gazebosim/gz-transport/pull/159.

## Steps to reproduce

Use Dockerfile:

```
FROM debian:bullseye

RUN apt-get update \
    && apt dist-upgrade -y \
    && apt install -y \
        git \
        wget \
        python3-pip \
        python-is-python3 \
        python3-distutils \
        build-essential \
        cmake \
        libtinyxml2-dev \
        libavutil-dev \
        libavcodec-dev \
        libavformat-dev \
        libavdevice-dev \
        libfreeimage-dev \
        libeigen3-dev \
        libgts-dev \
        libprotobuf-dev \
        libprotoc-dev \
        protobuf-compiler \
        protobuf-c-compiler \
        pkg-config \
        ruby \
        liburdfdom-dev \
        libx11-dev \
        libczmq-dev \
        libjsoncpp-dev \
        libcurl4-openssl-dev \
        libyaml-dev \
        libzip-dev

RUN python -m pip install vcstool
RUN python -m pip install colcon-common-extensions

RUN mkdir -p /ws/src
WORKDIR /ws/src
RUN wget https://raw.githubusercontent.com/ignition-tooling/gazebodistro/master/collection-fortress.yaml
RUN vcs import < collection-fortress.yaml
RUN colcon graph
RUN colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install
```

## Output

Getting errors such as:
```
/ws/src/gz-transport/src/NodeShared.cc: In member function ‘void ignition::transport::v11::NodeSharedPrivate::SecurityOnNewConnection()’:
/ws/src/gz-transport/src/NodeShared.cc:1606:23: error: ‘class zmq::socket_t’ has no member named ‘set’
 1606 |     this->subscriber->set(zmq::sockopt::plain_username, user);
      |                       ^~~
/ws/src/gz-transport/src/NodeShared.cc:1606:32: error: ‘zmq::sockopt’ has not been declared
 1606 |     this->subscriber->set(zmq::sockopt::plain_username, user);
      |                                ^~~~~~~
/ws/src/gz-transport/src/NodeShared.cc:1607:23: error: ‘class zmq::socket_t’ has no member named ‘set’
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,8 @@ if (UNIX AND NOT APPLE)
     message(WARNING "lsb_release executable not found. Disabling focal-specific workarounds")
   elseif (${RELEASE_CODENAME} STREQUAL "focal")
     set(UBUNTU_FOCAL 1)
+  elseif (${RELEASE_CODENAME} STREQUAL "bullseye")
+    set(DEBIAN_BULLSEYE 1)
   endif()
 endif()
 

--- a/include/ignition/transport/Helpers.hh
+++ b/include/ignition/transport/Helpers.hh
@@ -38,7 +38,8 @@
 // Avoid using deprecated set function when possible
 #if CPPZMQ_VERSION >= ZMQ_MAKE_VERSION(4, 7, 0)
   // Ubuntu Focal (20.04) packages a different "4.7.0"
-  #ifndef UBUNTU_FOCAL
+  // And so does apparently Debian Bullseye
+  #if !defined(UBUNTU_FOCAL) && !defined(DEBIAN_BULLSEYE)
     #define IGN_CPPZMQ_POST_4_7_0
   #endif
 #endif

--- a/include/ignition/transport/config.hh.in
+++ b/include/ignition/transport/config.hh.in
@@ -21,5 +21,6 @@
 
 #cmakedefine HAVE_IFADDRS 1
 #cmakedefine UBUNTU_FOCAL 1
+#cmakedefine DEBIAN_BULLSEYE 1
 
 #endif

